---
title: "Childhood gut microbiome is linked to mental health at school age via the functional connectome"
author: "Fran Querdasi"
post date: "2024-11-18"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup
## Set seed
```{r}
main.seed = 6024
```

## Load libraries
```{r}
suppressPackageStartupMessages({
  library(tidyverse)
  library(mixOmics)
  library(readxl)
  library(vegan)
  library(compositions)
  library(mosaic)
  library(Hmisc)
  library(rstatix)
  library(sjPlot)
  library(patchwork)
  library(corrplot)
  library(MASS) # for box-cox transformation
})
```

## Load data
```{r}
#load complete cases on brain and gut microbiome, and covariates
sample <- read_csv("../../data/analysis_datasets/collab_jess_fran/complete_gusto_postnatalmh_brain_microbiome_20221104.csv")

#matrix of brain networks (Y matrix)
brain <- read_csv( "../../data/analysis_datasets/collab_jess_fran/rsfc _values/6YO_network_rsfc_full.csv")

#child internalizing outcome variables
cbcl_y7 <- read_csv("../../data/child_beh/GUSTO_CBCL_scores_Y07_20220516.csv") %>% dplyr::select(
  subID = subjid,
  contains("int"),
  cbcltotprobtot_y7,
  cbcltotprobtscore_y7,
  cbcltotprobpct_y7
)

# cbcl Y7 raw internalizing variables to calculate alpha
cbcl_raw_y7 <- read_excel("../../data/child_beh/GUSTO_Y07_CBCL_20200330.xlsx", sheet = "Raw Data") %>% dplyr::select(
  subID = subjid,
  cbclq5_y7,
  cbclq14_y7,
  cbclq29_y7,
  cbclq30_y7,
  cbclq31_y7,
  cbclq32_y7,
  cbclq33_y7,
  cbclq35_y7,
  cbclq42_y7,
  cbclq45_y7,
  cbclq47_y7,
  cbclq49_y7,
  cbclq50_y7,
  cbclq51_y7,
  cbclq52_y7,
  cbclq54_y7,
  cbclq56a_y7,
  cbclq56b_y7,
  cbclq56c_y7,
  cbclq56d_y7,
  cbclq56e_y7,
  cbclq56f_y7,
  cbclq56g_y7,
  cbclq65_y7,
  cbclq69_y7,
  cbclq71_y7,
  cbclq75_y7,
  cbclq91_y7,
  cbclq102_y7,
  cbclq103_y7,
  cbclq111_y7,
  cbclq112_y7
)
``` 

## Merge data
```{r}
#left join brain to the sample file
brain_sample <- sample %>% left_join(brain, by = "subID")

#consolidate alpha diversity metrics
general_mb_path <- "../../data/microbiome/M24_M48/"
shannon <- read_tsv(str_c(general_mb_path, "diversity_tsvs/M2448_shannon.tsv")) %>% 
  dplyr::rename('#SampleID' = ...1)
richness <- read_tsv(str_c(general_mb_path, "diversity_tsvs/M2448_richness.tsv"))  %>% 
  dplyr::rename('#SampleID' = ...1)
evenness <- read_tsv(str_c(general_mb_path, "diversity_tsvs/M2448_evenness.tsv"))  %>% 
  dplyr::rename('#SampleID' = ...1)
faith <- read_tsv(str_c(general_mb_path, "diversity_tsvs/M2448_faith.tsv")) %>% 
  dplyr::rename('#SampleID' = ...1) 

alpha_div <- shannon %>% left_join(richness, by = "#SampleID") %>% left_join(evenness, by = "#SampleID") %>% left_join(faith, by = "#SampleID")

sample_ids <- sample %>% dplyr::select(subID, '#SampleID')

alpha_div_sample <- sample_ids %>% left_join(alpha_div, by = "#SampleID")

# merge clinical variables
metadata <- sample %>% left_join(cbcl_y7, by = "subID")

# write output files
write_csv(brain_sample, "../../data/analysis_datasets/collab_jess_fran/spls_data/brain_input_spls.csv")
write_csv(metadata, "../../data/analysis_datasets/collab_jess_fran/spls_data/clinvars_input_slsp.csv")
write_csv(alpha_div_sample,  "../../data/analysis_datasets/collab_jess_fran/spls_data/alpha_div_sample.csv")
```

# Data Preprocessing

## Create Genera and Taxonomy Combined File
```{r}
genus_otus_df = read_tsv("../../data/microbiome/M24_M48/input_files/genus_level/genus-table.tsv") #convert to df

#provide name for first column, OTU number (for merging)
genus_otus_df <- genus_otus_df %>% dplyr::rename(Feature.ID = `#OTU ID`)

#join taxonomy file to microbe table to get taxa names for genera
tax <- read.table("../../data/microbiome/M24_M48/input_files/M2448_taxonomy_edited.tsv", header=TRUE, sep = "\t")

#filter out any genera that have 0 abundance in all samples
genus_otus_df <-
  genus_otus_df %>% 
  mutate(
    otu_abundance = rowSums(.[2:671])
  )

# how many genera have 0 abundance in this subsample?
genus_otus_df %>% dplyr::count(otu_abundance==0) #30/353 genus-level taxa have 0 abundance across all samples

#filter out genera with zero abunadnce in all samples (first filtering step, we will do more filtering once the dataset has been transposed below)
genus_otus_df_filtered <- genus_otus_df %>% 
  filter(otu_abundance > 0) %>% 
  dplyr::select(-otu_abundance) #remove this column once you're done using it to filter

#transpose this taxonomy feature file so that taxonomy are columns
genus_otus_df_filtered_t <- as.data.frame(t(genus_otus_df_filtered))
colnames(genus_otus_df_filtered_t) <- as.character(genus_otus_df_filtered_t[1,])
genus_otus_df_filtered_t <- genus_otus_df_filtered_t[-1,]
genus_otus_df_filtered_t = setNames(data.frame(t(genus_otus_df_filtered[,-1])), genus_otus_df_filtered[,1])
#convert all character values for otu counts to numeric
genus_otus_df_filtered_test <- as.data.frame(lapply(genus_otus_df_filtered_t, as.numeric))
Sample_IDs <- rownames(genus_otus_df_filtered_t)

#name sample ID column for matching with metadata
genus_otus_df_filtered_test <- cbind(Sample_IDs, genus_otus_df_filtered_test) %>% dplyr::rename('#SampleID' = Sample_IDs)

#write new file combining taxonomy and genera counts as .tsv
tax_features_ra_genus_fp <- "../../data/analysis_datasets/collab_jess_fran/maaslin2/inputs/genus_tax_otus_ra.tsv"
write_tsv(genus_otus_df_filtered_test, tax_features_ra_genus_fp)
```

## Offset and CLR Transform Microbiome Data
Paper with special microbiome spls-da package and info on normalization to do beforehand: [https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0160169](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0160169), [http://mixomics.org/mixmc/](http://mixomics.org/mixmc/)

http://mixomics.org/mixmc/mixmc-preprocessing/ -- followed steps there
```{r}
#gut microbiome table -- can read in from this step to avoid doing everything above
otus <- read_tsv("../../data/analysis_datasets/collab_jess_fran/maaslin2/inputs/genus_tax_otus_ra.tsv")

#offset the data so there are no 0s
otus_offset <- otus %>% dplyr::select(-'#SampleID') + 1
ids <- otus %>% dplyr::select('#SampleID')

sum(which(otus_offset == 0))

#calculate non-zero samples for each otu (for checking which number to filer on makes sense)
non_zeros <- colSums(otus>0)
hist(non_zeros) #most are below 50

#pre-filtering to remove rare genera (keep genera for which sum of counts for that genera are above 5% of total counts in the entire dataset)
#(NOTE: doing this in whole dataset rather than subsample w brain data)
# tentatively decided on .05 as a cutoff 
low.count.removal <- function(
                        data, # genus count df of size n (sample) x p (genus)
                        percent=0.05 # cutoff chosen
                        ) 
  {
    keep.otu = which(colSums(data)*100/(sum(colSums(data))) > percent)
    data.filter = data[,keep.otu]
    return(list(data.filter = data.filter, keep.otu = keep.otu))
}

# call the function then apply on the offset data
result.filter <- low.count.removal(otus_offset, percent=0.05)
data.filter <- result.filter$data.filter

# check the number of variables kept after filtering
length(result.filter$keep.otu) #0.137 kept 50/353 features, 0.1385 keeps 50 features, .10 keeps 56 features, .05 keeps 64 features, .01 keeps 140 features

#apply CLR transformation
data.clr <- logratio.transfo(as.matrix(data.filter), 
                             logratio = 'CLR', offset = 0) 
#convert to dataframe
class(data.clr) <- "matrix"
data_clr <- as.data.frame(data.clr)

#add IDs back
data_clr_ids <- cbind(ids, data_clr)

#write filtered dataset
write_csv(data_clr_ids, "../../data/analysis_datasets/collab_jess_fran/spls_data/otus_spls_input.csv")
```

## Read in data if you don't want to redo steps above
```{r}
# load datasets (run these lines if you don't want to re-run the processing steps from above)
# gm data: data_clr_ids (64 gm features) -- add subID as id var, reduce to N=66 subsample
data_clr_subids <- read_csv("../../data/analysis_datasets/collab_jess_fran/spls_data/otus_spls_input.csv") %>% left_join(metadata %>% dplyr::select(`#SampleID`, subID), by = "#SampleID") %>% dplyr::filter(!is.na(subID)) %>% 
  dplyr::select(subID, everything(), -`#SampleID`)

mb_ids <- data_clr_subids %>% dplyr::select(subID)
mb_data_noids <- data_clr_subids %>% dplyr::select(-subID)

# brain data: brain_input_spls.csv (93 brain networks)
brain_input <- read_csv("../../data/analysis_datasets/collab_jess_fran/spls_data/brain_input_spls.csv")
brain_ids <- brain_input %>% dplyr::select(subID)
brain_data_noids <- brain_input %>% dplyr::select(-subID)

# clinical variables together: metadata 
metadata_clinical_cont <- read_csv("../../data/analysis_datasets/collab_jess_fran/spls_data/clinvars_input_slsp.csv") %>% 
  dplyr::select(
    subID,
    cbclintprobtot_y7
  )

meta_clin_cont_ids <- metadata_clinical_cont %>% dplyr::select(subID)

#full metadata dataframe, including covariates, but only in analytic subsample
metadata_sample <- metadata_clinical_cont %>% 
  dplyr::select(subID) %>% 
  left_join(metadata, by = "subID")
```

## Shorten taxa names
```{r}
# assign rownames (ids) to mb and brain data
rownames(mb_data_noids) <- mb_ids$subID
rownames(brain_data_noids) <- brain_ids$subID

#shorten mb names to be just genus (or family if genus not uniquely named) -- for plotting below
split_mb_names <- str_split(colnames(mb_data_noids), "\\.")

for (i in 1:length(colnames(mb_data_noids))){
    if (grepl("UCG|Ruminiclostridium", split_mb_names[[i]][length(split_mb_names[[i]])-1])) { #if the phylum is UCG, then take genus and species 
    colnames(mb_data_noids)[i] <- str_c(split_mb_names[[i]][length(split_mb_names[[i]])-1], split_mb_names[[i]][length(split_mb_names[[i]])], sep=".")
    }
  else if (grepl("__|[0-9]|uncultured|group", split_mb_names[[i]][length(split_mb_names[[i]])])) { #else if text after last dot (genus) is either a number or null or group...
    colnames(mb_data_noids)[i] <- split_mb_names[[i]][length(split_mb_names[[i]])-1] #then take the phylum as the name
    } 
    else {
    colnames(mb_data_noids)[i] <- split_mb_names[[i]][length(split_mb_names[[i]])] #if not, take the genus as the short name
  }
}

# write the data with shortened taxa names for later use
mb_data <- cbind(mb_ids, mb_data_noids)
write_csv(mb_data, "../../data/analysis_datasets/collab_jess_fran/spls_data/spls_mb_data_shortnames.csv")
```

# Analyses
## Brain --> Internalizing Symptoms sPLS1 models (X is matrix, Y is univariate)
Vignette for sPLS1 regression models: https://mixomicsteam.github.io/mixOmics-Vignette/id_04.html#id_04:spls1 

Note: data in spls are scaled by default 
```{r}
# select brain data in subsample that is complete on cbcl intprobs (55 ps) 
cbcl_subs <- metadata_sample %>% dplyr::filter(!is.na(cbclintprobtot_y7)) %>% dplyr::select(subID)
brain_cbcl <- cbcl_subs %>% left_join(brain_input, by = "subID")

brain_cbcl_ids <- brain_cbcl %>% dplyr::select(subID)
brain_cbcl_noids <- brain_cbcl %>% dplyr::select(-subID)

metadata_sample_cbcl <- metadata_sample %>% dplyr::filter(!is.na(cbclintprobtot_y7))

# initial sPLS model (should specify a large number of components first)
tune.spls1.int.brain <- pls(X=brain_cbcl_noids, Y=metadata_sample_cbcl$cbclintprobtot_y7, ncomp=4, mode='regression')

# use R^2 criterion to define the ideal number of dimensions/components (using repeated 10-fold cross validation)
set.seed(main.seed)
R2.spls1.int.brain <- perf(tune.spls1.int.brain, validation='Mfold',
                           folds=10, nrepeat=50)

plot(R2.spls1.int.brain, criterion = 'R2') # best is 2 components 

cbclintprobtot_y7 <- metadata_sample_cbcl %>% dplyr::select(cbclintprobtot_y7)
write_csv(brain_cbcl_noids, "X_input.csv")
write_csv(cbclintprobtot_y7, "y_input.csv")

# add a constant to all cbcl scores to avoid the error for tune.spls
rownames(brain_cbcl_noids) <- brain_cbcl_ids$subID
rownames(cbclintprobtot_y7) <- metadata_sample_cbcl$subID
cbclintprobtot_y7$cbclintprobtot_y7 <- cbclintprobtot_y7$cbclintprobtot_y7 + 1


# evaluate number of variables to select from X matrix (using number of components selected above)
# set up list of values
list.keepX <- c(5:10, seq(15, 50, 5)) # this should be thin at the start, and restricted to a small number of networks for a parsimonious model

# use R^2 criterion to define the ideal number of dimensions/components (using repeated 10-fold cross validation)
set.seed(main.seed)
tune.spls1.brain.win.r2 <- mixOmics::tune.spls(brain_cbcl_win_noids, cbclintprobtot_y7, ncomp= 2, 
                            test.keepX = list.keepX, 
                            validation = 'Mfold', 
                            folds = 10,
                            nrepeat = 100, 
                            progressBar = TRUE,
                            measure = 'R2') 

plot(tune.spls1.brain.win.r2) # best is 2 comps, first with 20 and second with 15 (like before) 

#how many components to keep w this?
choice.ncomp <- tune.spls1.brain.win.r2$choice.ncomp$ncomp #1
# how many x variables with 1 component? 5
choice.keepX <- tune.spls1.brain.win.r2$choice.keepX[1:choice.ncomp] #5 features
choice.keepX_2 <- tune.spls1.brain.win.r2$choice.keepX[1:2] #50 features if we specify 2 components
# for sake of parsimony, going with smaller number of components and features (not too much added gain)

# specify final model with 1 component and 5 brain features
spls1.int.brain <- spls(X=brain_cbcl_noids, Y=metadata_sample_cbcl$cbclintprobtot_y7, ncomp = choice.ncomp, keepX = choice.keepX, mode = "regression")

# extract list of features that were selected (each component is orthogonal)
selectVar(spls1.int.brain, comp = 1)$X$name 

# extract component scores
brain_cbcl <- as.data.frame(spls1.int.brain$variates$X) #outliers here are 010-21310 and 020-08045 

# what proportion of variance in internalizing is explained by the brain component?
spls1.int.brain$prop_expl_var$X #6% of the variance
tune.spls1.int.brain$prop_expl_var$X #39% of variance explained by component 1 (more than pre-tuning)

# plots of the results
#plot the component associated to the  X data set (here corresponding to a linear combination of the selected genes) vs. the component associated to the y variable (corresponding to the scaled y variable in PLS1 with one dimension)

plot(spls1.int.brain$variates$X, spls1.int.brain$variates$Y,
     xlab = 'X component', ylab = 'y component / scaled y') #looks like some high outliers on X (4-6)

cor(spls1.int.brain$variates$X, spls1.int.brain$variates$Y) #0.47
```

### Investigate and deal with brain outliers
2 brain outliers showed up in the scatterplot with the components for internalizing
```{r}
# based on the component scores from the brain-internalizing spls regression, ID 010-21310 and 020-08045  appear to be high multivariate outliers

brain_outliers <- brain_input %>% 
  dplyr::filter(
    subID == "010-21310" | subID == "020-08045"
  )

# get the ranges for all networks in the matrix
brain_means <- as.data.frame(colMeans(brain_input[2:94]))

# check distribution of each brain network -- on the whole, look relatively normal (some are slightly skewed)
hist.data.frame(brain_input)
# Calculate the variance of each variable
variances <- sapply(brain_input, var)

# Sort the variances from highest to lowest
sorted_variances <- as.data.frame(sort(variances, decreasing = TRUE))
colnames(sorted_variances) <- c("variance")
sorted_variances <- sorted_variances %>% rownames_to_column(var = "variable")

# Get the variable names corresponding to the sorted variances
sorted_variable_names <- names(sorted_variances)


# identify if the two multivariate outliers (010-21310 and 020-08045) are univariate outliers on any networks
brain_input %>% 
  pivot_longer(-subID) %>% #To make the data in long form required for `tidyverse`
  group_by(name) %>% #Based on which column you want to aggregate 
  identify_outliers(value) %>% 
  select(name, subID, value, is.outlier, is.extreme) %>% # extreme = Q3 + 3*IQR, outlier = Q3 + 1.5*IQR
  dplyr::filter(name %in% c("REW", "REW_MTL", "SAL_PMN", "SAL")) %>% 
  dplyr::filter(subID=="010-21310" | subID=="020-08045") #these are exterme outliers on REW and REW_MTL

# REW values are .8 and .63 (two highest values, next highest is .41), REW_MTL values are .76 and .69 (two highest values, next highest is .41)
```

### Create brain dataset with brain outliers winsorized to next highest values 
```{r}
# initialize a new dataset
brain_cbcl_win <- cbind(brain_cbcl_ids, brain_cbcl_noids)

# get the third highest REW value to replace the outliers with
top_5_REW_values <- brain_cbcl_win %>% 
  arrange(desc(REW)) %>% 
  slice_head(n=5) %>% 
  dplyr::select(REW)

top_5_REW_values$REW[3]

# replace both outlier sub's REW values with the third highest
brain_cbcl_win$REW[brain_cbcl_win$subID=="010-21310"] <- top_5_REW_values$REW[3]
brain_cbcl_win$REW[brain_cbcl_win$subID=="020-08045"] <- top_5_REW_values$REW[3]

# get the third higest REW_MTL values to replace outliers with
top_5_REW.MTL_values <- brain_cbcl_win %>% 
  arrange(desc(REW_MTL)) %>% 
  slice_head(n=5) %>% 
  dplyr::select(REW_MTL)

top_5_REW.MTL_values$REW_MTL[3]

brain_cbcl_win$REW_MTL[brain_cbcl_win$subID=="010-21310"] <- top_5_REW.MTL_values$REW_MTL[3]
brain_cbcl_win$REW_MTL[brain_cbcl_win$subID=="020-08045"] <- top_5_REW.MTL_values$REW_MTL[3]
```

### Tune the sPLS Brain --> Internalizing Symptoms models with the winsorized dataset
```{r}
brain_cbcl_win_noids <- brain_cbcl_win %>% dplyr::select(-subID)
rownames(brain_cbcl_win_noids) <- brain_cbcl_ids$subID

# initial sPLS model (should specify a large number of components first)
tune.spls1.int.brain.win <- pls(X=brain_cbcl_win_noids, Y=metadata_sample_cbcl$cbclintprobtot_y7, ncomp=4, mode='regression')

# use R^2 criterion to define the ideal number of dimensions/components (using repeated 10-fold cross validation)
set.seed(main.seed)
tune.spls1.brain.win.r2 <- mixOmics::tune.spls(brain_cbcl_win_noids, cbclintprobtot_y7, ncomp= 2, 
                            test.keepX = list.keepX, 
                            validation = 'Mfold', 
                            folds = 10,
                            nrepeat = 100, 
                            progressBar = TRUE,
                            measure = 'R2') 

plot(tune.spls1.brain.win.r2) # best is 2 comps, first with 20 and second with 15 (like before) 

#how many components to keep w this?
choice.ncomp.r2 <- tune.spls1.brain.win.r2$choice.ncomp$ncomp 
choice.keepX.r2 <- tune.spls1.brain.win.r2$choice.keepX[1:1] #20 features
choice.keepX_2.r2 <- tune.spls1.brain.win.r2$choice.keepX[1:2] #20 features comp1, 15 comp2 

# specify final model with same number of features as before 
spls1.int.brain.win <- spls(X=brain_cbcl_win_noids, Y=metadata_sample_cbcl$cbclintprobtot_y7, ncomp = 2, keepX = choice.keepX_2.r2, mode = "regression")

# extract list of features that were selected (each component is orthogonal)
selectVar(spls1.int.brain.win, comp = 1)$X$name #  "REW"     "SAL_PMN" "REW_MTL" "SAL"     "MTL"     "SAL_SMD" "CON_PMN" "SAL_AUD" "VAN_PMN" "DMN_SAL" "SAL_SML" "SAL_VAN" "MTL_PMN" "DMN_MTL" "CON_SMD" "DMN_REW" "SAL_DAN" "DAN_VIS" "VIS_SMD" "SAL_CON"
selectVar(spls1.int.brain.win, comp = 2)$X$name #"VIS_REW" "DMN_REW" "VAN_SMD" "FPN_REW" "VAN_REW" "CON_REW" "SMD_SML" "DAN_SML" "VAN_MTL" "DAN_VAN" "REW_PMN" "DMN"     "VIS_PMN" "DAN_REW" "DMN_CON"

# plot the loadings for each feature
brain.int.win_loadings_c1 <- selectVar(spls1.int.brain.win, comp=1)$X$value %>% rownames_to_column(var = "network")
colnames(brain.int.win_loadings_c1)[2] <- "loading_C1"

brain.int.win_loadings_c2 <- selectVar(spls1.int.brain.win, comp=2)$X$value %>% rownames_to_column(var = "network")
colnames(brain.int.win_loadings_c2)[2] <- "loading_C2"

# determine the variable importance (loading magnitude weighted by variance explained?)
vip.spls1.int.brain.win <- vip(spls1.int.brain.win)
vip.spls1.int.brain.win[selectVar(spls1.int.brain.win, comp=1)$X$name, 1] # 10 variables with VIP above 1 in comp 1
vip.spls1.int.brain.win[selectVar(spls1.int.brain.win, comp=2)$X$name, 2] # 8 variables with VIP above 1 in comp 2

# what proportion of variance in microbiome is explained by each component
spls1.int.brain.win$prop_expl_var$X #17% explained by comp 1, 32% by comp 2
spls1.int.brain.win$prop_expl_var$Y

# plots of the results
plot(spls1.int.brain.win$variates$X, spls1.int.brain.win$variates$Y,
     xlab = 'X component', ylab = 'y component / scaled y') #

cor(spls1.int.brain.win$variates$X, spls1.int.brain.win$variates$Y) #comp1 x and y are correlated at 47, comp2 at .38

# extract component scores
brain.int.win.comp_scores <- as.data.frame(spls1.int.brain.win$variates$X) %>% 
  dplyr::rename(brain_int_win_comp1 = comp1,
                brain_int_win_comp2 = comp2) %>% 
  rownames_to_column(var="subID")

# save the loadings and vips, combining component 1 and component 2
brain_loadings <- selectVar(spls1.int.brain.win, comp=1)$X$value %>% rownames_to_column(var = "network") 
colnames(brain_loadings) <- c("network", "loading")
brain_vip<- as.data.frame(vip.spls1.int.brain.win[selectVar(spls1.int.brain.win, comp=1)$X$name, 1]) %>% rownames_to_column(var="network")
colnames(brain_vip) <- c("network", "vip")

brain_load_vip_c1 <- brain_loadings %>% left_join(brain_vip, by = "network") %>% dplyr::mutate(loading = signif(loading, 2), vip = signif(vip, 2), signature = "SOFA Intra-Network")


brain_loadings_c2 <- selectVar(spls1.int.brain.win, comp=2)$X$value %>% rownames_to_column(var = "network") 
colnames(brain_loadings_c2) <- c("network", "loading")

brain_vip_c2 <- as.data.frame(vip.spls1.int.brain.win[selectVar(spls1.int.brain.win, comp=2)$X$name, 2]) %>% rownames_to_column(var="network")
colnames(brain_vip_c2) <- c("network", "vip")

brain_load_vip_c2 <- brain_loadings_c2 %>% left_join(brain_vip_c2, by = "network") %>% dplyr::mutate(loading = signif(loading, 2), vip = signif(vip, 2), signature = "SOFA Inter-Network")

brain_load_vip_comb <- brain_load_vip_c1 %>% bind_rows(brain_load_vip_c2) %>% dplyr::select(network, signature, everything())
write_csv(brain_load_vip_comb, "tables/TableS1.csv")
```

### Plot Final Brain --> Internalizing Symptoms Loadings
```{r}
# plot loadings using dataframes for the added flexibility of ggplot
# create a factor that sorts the loadings by magnitude 
brain.int.win_loadings_c1$network <- factor(brain.int.win_loadings_c1$network, levels = brain.int.win_loadings_c1$network[order(brain.int.win_loadings_c1$loading_C1, decreasing = TRUE)])

brain.int.win_loadings_c2$network <- factor(brain.int.win_loadings_c2$network, levels = brain.int.win_loadings_c2$network[order(brain.int.win_loadings_c2$loading_C2, decreasing = TRUE)])

# wherever there is "REW", replace with "SOFA" (more accurate name actually used in Seitzman 2020)
brain.int.win_loadings_c1$network <- str_replace(brain.int.win_loadings_c1$network, "REW", "SOFA")
brain.int.win_loadings_c2$network <- str_replace(brain.int.win_loadings_c2$network, "REW", "SOFA")

# graph the 10 highest loadings by magnitude, with vip > 1
brain_c1 <- ggplot(brain.int.win_loadings_c1 %>% slice_max(., n=10, order_by=abs(loading_C1)), aes(x = reorder(network, loading_C1), y = loading_C1)) +
  geom_col(fill = '#95cacb') + 
  coord_flip() +
  ylab("Loading") + xlab("Brain Network") +
  labs(tag = "A") + 
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=14), axis.text.y = element_text(color="black", size=14), axis.title = element_text(size=14), plot.tag=element_text(size=14)) +
  ggtitle("SOFA, MTL, SAL Intra-
  Network Signature") +
  theme(plot.title = element_text(hjust=0.5, size=14, face='bold'))

# graph comp2 loadings iwth vip > 1
brain_c2 <- ggplot(brain.int.win_loadings_c2 %>% slice_max(., n=8, order_by=abs(loading_C2)), aes(x = reorder(network, loading_C2), y = loading_C2)) +
  geom_col(fill = '#fd988d') + 
  coord_flip() +
  ylab("Loading") + xlab("Brain Network") +
  labs(tag = "C") + 
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=14), axis.text.y = element_text(color="black", size=14), axis.title = element_text(size=14), plot.tag=element_text(size=14)) +
  ggtitle("SOFA Inter-Network 
  Signature") +
  theme(plot.title = element_text(hjust=0.5, size=14, face='bold'))

# use patchwork package to arrange the plots next to each other
brain_c1 + brain_c2

ggsave('figures/Figure1.png', brain_c1 + brain_c2, width = 9, height=5, dpi=1000)
```

### Get brain network means, variances, and range
```{r}
# calculate means and sds 
brain_means <- brain_cbcl_win_noids %>% 
  summarise_all(mean)

brain_sds <- brain_cbcl_win_noids %>% 
  summarise_all(sd)

brain_mins <- brain_cbcl_win_noids %>% 
  summarise_all(min)

brain_maxes <- brain_cbcl_win_noids %>% 
  summarise_all(max)

# reshape the dataframe
means <- brain_means %>% pivot_longer(
  cols = everything(),
  names_to = 'network',
  values_to = 'mean'
)

sds <- brain_sds %>% pivot_longer(
  cols = everything(),
  names_to = 'network',
  values_to = 'sd'
) 

mins <- brain_mins %>% pivot_longer(
  cols = everything(),
  names_to = 'network',
  values_to = 'min'
) 

maxes <- brain_maxes %>% pivot_longer(
  cols = everything(),
  names_to = 'network',
  values_to = 'max'
) 

summary <- means %>% full_join(sds, by = "network") %>% full_join(mins, by = "network") %>% full_join(maxes, by = "network")

write_csv(summary, "brain_network_descriptives.csv")
```

### Read in additional (diet) covariates
```{r}
# merge component scores with rest of dataset
metadata_comp <- metadata %>% left_join(brain.int.win.comp_scores, by = "subID") 

diet <- read_csv("../../data/mb_covariates/diet_covariates.csv") %>% 
  dplyr::rename(subID = subjid)

# correlations between diet covariates
diet_m <- diet %>% dplyr::select(-subID)
testRes = cor.mtest(diet_m, conf.level = 0.95)
M = cor(diet_m)
corrplot(M, p.mat = testRes$p, sig.level = 0.05, order = 'hclust', addrect = 2)
# some clusters: CHO and fat (-), DQI, protein, fiber (+)

# correlation with brain component scores and alpha div
diet_omics_m <- metadata_comp %>% dplyr::select(subID, contains("win"), cbclintprobtot_y7, any_bf_months) %>% left_join(alpha_div_sample, by = "subID") %>% left_join(diet, by = "subID") %>% dplyr::select(-`#SampleID`, -subID) %>% drop_na() %>% dplyr::select(-contains("DQI"), -any_bf_months) %>% dplyr::rename(
  "SOFA, MTL, SAL Intra-Network Signature" = brain_int_win_comp1,
  "SOFA Inter-Network Signature" = brain_int_win_comp2,
  "Internalizing Symptoms" = cbclintprobtot_y7,
  "Shannon Index" = shannon_entropy,
  "Observed Features" = observed_features,
  "Pielou Evenness" = pielou_evenness,
  "Faith's PD" = faith_pd,
  "Protein" = `Protein_%energy`,
  "Total Fat" = `Fat_%energy`,
  "Carbs" = `CHO_%energy`,
  "Fiber" = Fibre.per.1000kcal,
  "Saturated Fat" = `SatFat_%`,
  "MUFA" = `MUFA_%`,
  "PUFA" = `PUFA_%`
)

testRes1 = cor.mtest(diet_omics_m, conf.level = 0.95)
M1 = cor(diet_omics_m)
# only significant association with alpha div is fiber + related to shannon entropy and observed features, brain comp2 is - related to fat % energy and + related to cho % energy, internalizing is + related to PUFA

corrplot(M1,
         method="color",
         type="lower",
         diag=TRUE,
         p.mat = testRes1$p,
         insig = "label_sig",
         sig.level = c(.001, .01, .05),
         pch.cex = 0.8,
         pch.col = "yellow",
         tl.col="black",
         tl.cex=0.8,
         addCoef.col = "black",
         tl.pos="l",
         cl.pos="r",
         outline=TRUE) 
# can't figure out how to save this besides taking a picture of output

cor.test(diet_m$`Fat_%energy`, diet_m$Fibre.per.1000kcal) # cor is -0.25 (not too strong to put both in regression)
cor.test(diet_m$`Fat_%energy`, diet_m$`CHO_%energy`) #cor is -0.86 -- collinear, choose 1 to put in regression (fat has higher magnitude of correlation with brain comp so put that one)
cor.test(diet_m$`Fat_%energy`, diet_m$`PUFA_%`) # cor is -0.28 
cor.test(diet_m$Fibre.per.1000kcal, diet_m$`PUFA_%`) # very small correlation 

# add dataset with standard covariates
braincomp_alpha_covs <- metadata_comp %>% dplyr::select(-observed_features, (-contains("comp") & !contains("noamyg"))) %>% left_join(alpha_div_sample, by = "subID") %>% left_join(diet, by = "subID") %>% dplyr::select(-contains("Sample")) 

# make bf into ordered factor
braincomp_alpha_covs$any_bf_months <- factor(braincomp_alpha_covs$any_bf_months, levels = c("1M_to_3M", "3M_to_6M", "6M_to_12M", ">12M"))

# breastfeeding associations with brain/int components, internalizing, or alpha diversity: anovas
summary(aov(brain_int_win_comp1~any_bf_months, data=braincomp_alpha_covs)) 
summary(aov(brain_int_win_comp2~any_bf_months, data=braincomp_alpha_covs)) #ns
summary(aov(cbclintprobtot_y7~any_bf_months, data=braincomp_alpha_covs)) #ns
summary(aov(shannon_entropy~any_bf_months, data=braincomp_alpha_covs))
summary(aov(observed_features.x~any_bf_months, data=braincomp_alpha_covs))
summary(aov(pielou_evenness~any_bf_months, data=braincomp_alpha_covs))
summary(aov(faith_pd~any_bf_months, data=braincomp_alpha_covs))

# is breastfeeding correlated with any diet vars we have selecte?
summary(aov(`PUFA_%`~any_bf_months, data=braincomp_alpha_covs)) # 
summary(aov(`Fat_%energy`~any_bf_months, data=braincomp_alpha_covs)) #ns
summary(aov(Fibre.per.1000kcal  ~any_bf_months, data=braincomp_alpha_covs)) #ns

# birth method associations with brain/int components, internalizing, or alpha diversity: t-tests
t.test(braincomp_alpha_covs$deliv_mode, braincomp_alpha_covs$brain_int_noamyg_win_comp1) # ns
t.test(braincomp_alpha_covs$deliv_mode, braincomp_alpha_covs$brain_int_noamyg_win_comp2)
t.test(braincomp_alpha_covs$deliv_mode, braincomp_alpha_covs$cbclintprobtot_y7) #  

# is birth method associated with any diet vars we have selected?
t.test(braincomp_alpha_covs$deliv_mode, braincomp_alpha_covs$`PUFA_%`)

# how many infants are in each breastfeeding category?
braincomp_alpha_covs %>% dplyr::filter(!is.na(cbclintprobtot_y7)) %>% dplyr::count(any_bf_months) # roughly 1/4 were still gettin breastmilk after a year

# is GA and BW related to each other?
cor.test(braincomp_alpha_covs$GA_centered, braincomp_alpha_covs$BW_centered) # yes

# we want to add diet covariates to the regression with brain and internalizing, bc 1) Fat is related to brain and PUFA is related to internalizing, 2) we want to keep largely same covariate set for each analysis
```

### Missing date rates for covariates
```{r}
# get analytic sample with cbcl, brain, and gm
sample_braincomp_alpha_covs <- braincomp_alpha_covs %>% dplyr::filter(!is.na(cbclintprobtot_y7))

# count number missing on each covariate: ga, bw, sex, delivery mode, diet, breastfeeding
sample_braincomp_alpha_covs %>% dplyr::count(is.na(sex_centered)) # complete
sample_braincomp_alpha_covs %>% dplyr::count(is.na(GA_centered)) # complete
sample_braincomp_alpha_covs %>% dplyr::count(is.na(BW_centered)) # complete
sample_braincomp_alpha_covs %>% dplyr::count(is.na(deliv_mode_str)) # complete
sample_braincomp_alpha_covs %>% group_by(is.na(Fibre.per.1000kcal)) %>% summarise(n = n()) %>%
  mutate(freq = n / sum(n)) #43/55 have diet data (78%)
sample_braincomp_alpha_covs %>% dplyr::count(is.na(any_bf_months)) # 53/55 (4% missing)
```

### Transform the internalizing symptom outcome variable for linear regression
Box-cox transformation in R: https://www.r-bloggers.com/2022/10/box-cox-transformation-in-r/ 
```{r}
# distribution of component scores
gf_histogram(~brain.int.win.comp_scores$brain_int_win_comp1, bins=15) #normal enough, a few outliers
gf_histogram(~brain.int.win.comp_scores$brain_int_win_comp2, bins=15) #skewed
gf_histogram(~metadata_comp$cbclintprobtot_y7) # skewed; do box-cox transformation 

# box-cox transformation on intprob (add constant of 1 to all values to make it positive, required for box-cox)
metadata_comp <- metadata_comp %>% dplyr::mutate(cbclintprobtot_y7_pos = cbclintprobtot_y7+1)
# extract optimal lambda 
boxcox(lm(metadata_comp$cbclintprobtot_y7_pos ~ 1),
       lambda = seq(-2, 2, 1/10), 
       plotit = TRUE,  
       eps = 1/50,     
       xlab = expression(lambda), 
       ylab = "log-Likelihood",   
       ) #95% CI is roughly 0 to 0.5

b <- boxcox(lm(metadata_comp$cbclintprobtot_y7_pos ~ 1),
       lambda = seq(-2, 2, 1/10), 
       plotit = TRUE,  
       eps = 1/50,     
       xlab = expression(lambda), 
       ylab = "log-Likelihood",   
       )

lambda <- b$x[which.max(b$y)] # lambda = 0.26
 
# apply the transformation 
metadata_comp <- metadata_comp %>%
  dplyr::mutate(
    cbclintprobtot_y7_pos_boxcox = (cbclintprobtot_y7_pos ^ lambda- 1)/lambda 
  )

#graph transformed varaible
gf_histogram(~metadata_comp$cbclintprobtot_y7_pos_boxcox) #looks slightly better
```

### Brain Signatures --> Internalizing Regression Model, controlling for selected covariates
```{r}
# regression with comp scores predicting internalizing, controlling for covariates
comp1_int_trans_diet <- lm(cbclintprobtot_y7_pos_boxcox ~ brain_int_win_comp1 + sex_centered + BW_centered + deliv_mode + `PUFA_%` + `Fat_%energy` + meanFD_centered, data=braincomp_alpha_covs)
tab_model(comp1_int_trans_diet, robust=TRUE, show.std=TRUE, show.se=TRUE) # comp1 is positively related to internalizing

comp2_int_trans <- lm(cbclintprobtot_y7_pos_boxcox ~ brain_int_win_comp2 + sex_centered + GA_centered + BW_centered + deliv_mode + `PUFA_%` + `Fat_%energy` + meanFD_centered, data=braincomp_alpha_covs)
tab_model(comp2_int_trans, robust=TRUE, show.std=TRUE, show.se=TRUE) # comp2 is positively related to internalizing
```

## Brain Signatures Associations with Microbiome Variables

### Alpha Diversity
Alpha diversity + covs ~ brain comps
```{r}
# shannon
shannon_comp1 <- lm(brain_int_win_comp1 ~ shannon_entropy + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
#assumptions for comp1 models (will look similar for other metrics bc alpha divs are all pretty normally distributed)
shannon_comp1_resids <- resid(shannon_comp1)
shannon_comp1_fitted <- fitted(shannon_comp1)
##normality of residuals
hist(shannon_comp1_resids) # relatively normal, a little bit sparse upper part of distributino
##heteroskedasticity + linearity
plot(shannon_comp1_resids ~ shannon_comp1_fitted) #

tab_model(shannon_comp1, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns

shannon_comp1_ncov <- lm(brain_int_win_comp1 ~ shannon_entropy, data = braincomp_alpha_covs)
tab_model(shannon_comp1_ncov, robust=TRUE, show.std=TRUE) #ns

shannon_comp2 <- lm(brain_int_win_comp2 ~ shannon_entropy + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)

#assumptions for comp2 models (will look similar for other metrics bc alpha divs are all pretty normally distributed)
shannon_comp2_resids <- resid(shannon_comp2)
shannon_comp2_fitted <- fitted(shannon_comp2)
##normality of residuals
hist(shannon_comp2_resids) # pretty normal
##heteroskedasticity + linearity
plot(shannon_comp2_resids ~ shannon_comp2_fitted) # 

tab_model(shannon_comp2, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
shannon_comp1_ncov <- lm(brain_int_win_comp1 ~ shannon_entropy, data = braincomp_alpha_covs)
tab_model(shannon_comp1_ncov, robust=TRUE, show.std=TRUE) #ns

# observed features
feat_comp1 <- lm(brain_int_win_comp1 ~ observed_features.x + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(feat_comp1, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
feat_comp1_ncovs <- lm(brain_int_win_comp1 ~ observed_features.x, data = braincomp_alpha_covs)
tab_model(feat_comp1_ncovs, robust=TRUE, show.std=TRUE) #ns

feat_comp2 <- lm(brain_int_win_comp2 ~ observed_features.x + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(feat_comp2, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
feat_comp2_ncovs <- lm(brain_int_win_comp2 ~ observed_features.x, data = braincomp_alpha_covs)
tab_model(feat_comp2_ncovs, robust=TRUE, show.std=TRUE) #ns

# faith pd
faith_comp1 <- lm(brain_int_win_comp1 ~ faith_pd + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(faith_comp1, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
faith_comp1_ncovs <- lm(brain_int_win_comp1 ~ faith_pd, data = braincomp_alpha_covs)
tab_model(faith_comp1_ncovs, robust=TRUE, show.std=TRUE) #ns

faith_comp2 <- lm(brain_int_win_comp2 ~ faith_pd + sex_centered + GA_centered + BW_centered + deliv_mode +`Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(faith_comp2, robust=TRUE, show.std=TRUE, show.se=TRUE) #sig, higher faith with higher comp2 scores (beta = .38, p=.004)
faith_comp2_ncovs <- lm(brain_int_win_comp2 ~ faith_pd, data = braincomp_alpha_covs)
tab_model(faith_comp2_ncovs, robust=TRUE, show.std=TRUE) #sig, same as with covariates

# evenness
even_comp1 <- lm(brain_int_win_comp1 ~ pielou_evenness + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(even_comp1, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
even_comp1_ncovs <- lm(brain_int_win_comp1 ~ pielou_evenness, data = braincomp_alpha_covs)
tab_model(even_comp1_ncovs, robust=TRUE, show.std=TRUE) #ns

even_comp2 <- lm(brain_int_win_comp2 ~ pielou_evenness + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_alpha_covs)
tab_model(even_comp2, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns
even_comp2_ncovs <- lm(brain_int_win_comp2 ~ pielou_evenness, data = braincomp_alpha_covs)
tab_model(even_comp2_ncovs, robust=TRUE, show.std=TRUE) #ns

# quick scatterplot of faith and brain comp2
ggplot(braincomp_alpha_covs, aes(faith_pd, brain_int_win_comp2)) +
  geom_jitter() +
  geom_smooth(method="lm")

#note: did not remove or winsorize high outliers on Faith bc if you look in the full sample, it appears Faith has bimodal distribution (there is a smaller high group which is just very small in this subsample) 
```

### Figure 2: Faith's PD and SOFA Inter-Network Signature
We will residualize both x and y to represent a partial regression coefficient, which is what we get from a MLR controlling for covariates. 
```{r}
braincomp_faith <- braincomp_alpha_covs %>% 
  filter(!is.na(sex_centered) & !is.na(GA_centered) &!is.na(BW_centered) &!is.na(deliv_mode) & !is.na(`Fat_%energy`) & !is.na(Fibre.per.1000kcal) & !is.na(meanFD_centered) & !is.na(cbclintprobtot_y7) & !is.na(faith_pd))

# residualize brain comp2 scores by covariates
braincomp2_resid <- lm(brain_int_win_comp2 ~sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_faith)

# resiaulize faith by covariates
faith_resid <- lm(faith_pd ~sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + meanFD_centered, data = braincomp_faith)

braincomp_faith <- braincomp_faith %>% 
  mutate(
    braincomp2_resid = resid(braincomp2_resid),
    faith_braincomp2_resid = resid(faith_resid)
  ) 

# graph
ggplot(braincomp_faith, aes(faith_braincomp2_resid, braincomp2_resid)) +
  geom_point(position=position_jitter(w=0.3, h=0)) +
  geom_smooth(method="lm", fill = '#7570b3', color = '#7570b3') +
  ylab("SOFA Inter-Network \n Connectivity Brain Signature") + xlab("Faith's Phylogenetic Diversity") +
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12), axis.title = element_text(size=15)) 

# save
ggsave("figures/Figure4.jpg", width = 9, height=5, dpi=1000)
```

### Microbial Profile(s) with SOFA, MTL, SAL Intra-Network Signature
```{r}
# grab mb ids with values for the brain-int compoment scores
mb_data_brainint <- brain.int.win.comp_scores %>% dplyr::select(subID) %>% left_join(mb_data, by = "subID") #%>% dplyr::select(-subID)
rownames(mb_data_brainint) <- mb_data_brainint$subID
mb_data_brainint <- mb_data_brainint %>% dplyr::select(-subID)

# initial sPLS model (should specify a large number of components first)
tune.spls1.braincomp.mb <- pls(X=mb_data_brainint, Y=brain.int.win.comp_scores$brain_int_win_comp1, ncomp=4, mode='regression')

# evaluate number of variables to select from X matrix (using number of components selected above)
list.keepX <- c(5:10, seq(15, 93, 5)) # specify selection up to the number of features in the dataset (93)

# evaluate tuning w/ R^2
set.seed(main.seed)
tune.spls1.braincomp.mb.R2 <- mixOmics::tune.spls(mb_data_brainint, brain.int.win.comp_scores$brain_int_win_comp1, ncomp= 2, 
                            test.keepX = list.keepX, 
                            validation = 'Mfold', 
                            folds = 10,
                            nrepeat = 100, 
                            progressBar = TRUE,
                            measure = 'R2') 

plot(tune.spls1.braincomp.mb.R2) #highest R^2 (0.02) is with 1 component and 5 features

#how many components to keep w this?
choice.ncomp <- tune.spls1.braincomp.mb.R2$choice.ncomp$ncomp #1
# how many x variables with 1 component? 5
choice.keepX <- tune.spls1.braincomp.mb.R2$choice.keepX[1:1] #5

# specify final model with 1 components, 5 features on comp1
spls1.braincomp.mb <- spls(X=mb_data_brainint, Y=brain.int.win.comp_scores$brain_int_win_comp1, ncomp = choice.ncomp, keepX = choice.keepX, mode = "regression")

# extract list of features that were selected (each component is orthogonal)
selectVar(spls1.braincomp.mb, comp = 1)$X$name #selected "Eubacterium" "hallii" "Coprococcus" "Dialister"   "Weissella"  
plotLoadings(spls1.braincomp.mb, 'X') # positive is Eubacterium, blautia; negative is hallii, coproccocus, and dialister

plot(spls1.braincomp.mb$variates$X, spls1.braincomp.mb$variates$Y,
     xlab = 'X component', ylab = 'y component / scaled y') #

cor(spls1.braincomp.mb$variates$X, spls1.braincomp.mb$variates$Y) # cor = .43

# extract component scores to investigate low outlier (010-04087)
mbXbraincomp_comp1 <- as.data.frame(spls1.braincomp.mb$variates$X) %>% 
  dplyr::rename(mb_braincomp_comp1 = comp1) %>% rownames_to_column(var = "subID")

mb_data_brainint %>% 
  rownames_to_column(var = "subID") %>% 
  pivot_longer(-subID) %>% #To make the data in long form required for `tidyverse`
  group_by(name) %>% #Based on which column you want aggregate 
  identify_outliers(value) %>% 
  select(name, subID, value, is.outlier, is.extreme) %>% # extreme = Q3 + 3*IQR, outlier = Q3 + 1.5*IQR
  dplyr::filter(name %in% c("Eubacterium", "hallii", "Coprococcus", "Dialister", "Weissella")) %>% 
  dplyr::filter(subID=="010-04087") # this sub is extreme high outlier on coprococcus, hallii, weissella, and high outlier on dialister
```

### Winsorize low microbiome outlier
```{r}
# initialize a new dataset
mb_data_brainint_win <- mb_data_brainint %>% rownames_to_column(var = "subID")

# get the second highest coproccocus value to replace the outlier with (based on visual inspection, second highest value is ok)
top_2_coproccocus_values <- mb_data_brainint %>% 
  arrange(desc(Coprococcus)) %>% 
  slice_head(n=2) %>% 
  dplyr::select(Coprococcus)

top_2_coproccocus_values$Coprococcus[2]

# replace both outlier sub's Coprococcus value with the second highest
mb_data_brainint_win$Coprococcus[mb_data_brainint_win$subID=="010-04087"] <- top_2_coproccocus_values$Coprococcus[2]

# same procedure but for hallii
top_2_hallii_values <- mb_data_brainint %>% 
  arrange(desc(hallii)) %>% 
  slice_head(n=2) %>% 
  dplyr::select(hallii)

top_2_hallii_values$hallii[2]

mb_data_brainint_win$hallii[mb_data_brainint_win$subID=="010-04087"] <- top_2_hallii_values$hallii[2]

# same procedure for weissella
top_2_weissella_values <- mb_data_brainint %>% 
  arrange(desc(Weissella)) %>% 
   slice_head(n=2) %>% 
  dplyr::select(Weissella)

top_2_weissella_values$Weissella[2]

mb_data_brainint_win$Weissella[mb_data_brainint_win$subID=="010-04087"] <- top_2_weissella_values$Weissella[2]


# convert subID column to rowname for reading into spls
ids <- mb_data_brainint_win %>% dplyr::select(subID)
mb_data_brainint_win <- mb_data_brainint_win %>% dplyr::select(-subID) 
rownames(mb_data_brainint_win) <- ids$subID
```

### Microbial Pattern(s) with SOFA, MTL, SAL Intra-Network Signature, low outlier winsorized
```{r}
# initial sPLS model (should specify a large number of components first)
tune.spls1.braincomp.mbwin <- pls(X=mb_data_brainint_win, Y=brain.int.win.comp_scores$brain_int_win_comp1, ncomp=4, mode='regression')

# evaluate number of variables to select from X matrix (using number of components selected above)
list.keepX <- c(5:10, seq(15, 93, 5)) # specify selection up to the number of features in the dataset (93)

# evaluate tuning w/ R^2
set.seed(main.seed)
tune.spls1.braincomp.mbwin.R2 <- mixOmics::tune.spls(mb_data_brainint_win, brain.int.win.comp_scores$brain_int_win_comp1, ncomp= 2, 
                            test.keepX = list.keepX, 
                            validation = 'Mfold', 
                            folds = 10,
                            nrepeat = 100, 
                            progressBar = TRUE,
                            measure = 'R2') 

plot(tune.spls1.braincomp.mb.R2) #highest R^2 (0.02) is with 1 component and 5 features

#how many components to keep w this?
choice.ncomp <- tune.spls1.braincomp.mbwin.R2$choice.ncomp$ncomp #1
# how many x variables with 1 component? 5
choice.keepX <- tune.spls1.braincomp.mbwin.R2$choice.keepX[1:1] #5

# specify final model with 1 components, 5 features on comp1
spls1.braincomp.mbwin <- spls(X=mb_data_brainint_win, Y=brain.int.win.comp_scores$brain_int_win_comp1, ncomp = choice.ncomp, keepX = choice.keepX, 
                    mode = "regression")

# extract list of features that were selected (each component is orthogonal)
selectVar(spls1.braincomp.mbwin, comp = 1)$X$name #selected "Eubacteriaceae" "Anaerobutyricum hallii" "Weissella" "Dialister"   "coproccocus"  
plotLoadings(spls1.braincomp.mbwin, 'X') # positive is Eubacteriaceae; negative is Anaerobutyricum hallii, coproccocus, weissella, and dialister

selectVar(spls1.braincomp.mbwin, comp = 1)$X$value

plot(spls1.braincomp.mbwin$variates$X, spls1.braincomp.mbwin$variates$Y,
     xlab = 'X component', ylab = 'y component / scaled y') #

cor(spls1.braincomp.mbwin$variates$X, spls1.braincomp.mbwin$variates$Y) # cor = .51
spls1.braincomp.mbwin$prop_expl_var$X

# determine the variable importance (loading magnitude weighted by variance explained?)
vip.spls1.braincomp.mbwin <- vip(spls1.braincomp.mbwin)
vip.spls1.braincomp.mbwin[selectVar(spls1.braincomp.mbwin, comp=1)$X$name, 1] # 3 variables with VIP above 1 in comp 1

# extract scores for mb component associated with brain comp2
braincomp1.mbwin <- as.data.frame(spls1.braincomp.mbwin$variates$X) %>% 
  dplyr::rename(mb_braincomp1 = comp1) %>% 
  rownames_to_column(var="subID")
```

### Graph Microbe Loadings for SOFA, MTL, SAL Intra-Network Signature
```{r}
# extract feature loadings
braincomp1_mb <- selectVar(spls1.braincomp.mbwin, comp = 1)$X$value %>% rownames_to_column(var = "genus")
colnames(braincomp1_mb)[2] <- "loading_C1"

# rename hallii to updated name (Anaerobutyricum hallii)
braincomp1_mb$genus[braincomp1_mb$genus == "hallii"] <- "Anaerobutyricum hallii" 
braincomp1_mb$genus[braincomp1_mb$genus == "Eubacterium"] <- "Eubacteriaceae"

# create a factor that sorts the loadings by magnitude 
braincomp1_mb$genus <- factor(braincomp1_mb$genus, levels = braincomp1_mb$genus[order(braincomp1_mb$loading_C1, decreasing = TRUE)])

# graph the 3 highest loadings (those with VIP >1 )by magnitude
mb_pattern1 <- ggplot(braincomp1_mb %>% slice_max(., n=3, order_by=abs(loading_C1)), aes(x = reorder(genus, loading_C1), y = loading_C1)) +
  geom_col(fill = '#95cacb') + 
  coord_flip() +
  ylab("Loading") + xlab("Genus") +
  labs(tag = "B") +
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12), axis.title = element_text(size=16), plot.margin = margin(10, 10, 10, 10), plot.tag=element_text(size=14)) +
  ggtitle("Microbial \n Profile 1") +
  theme(plot.title = element_text(hjust=0.5, size=14, face='bold'))
```

### Microbial Profile(s) with SOFA Inter-Network Signature
```{r}
# initial sPLS model (should specify a large number of components first)
tune.spls1.braincomp2.mb <- pls(X=mb_data_brainint, Y=brain.int.win.comp_scores$brain_int_win_comp2, ncomp=4, mode='regression')

# evaluate number of variables to select from X matrix (using number of components selected above)
list.keepX <- c(5:10, seq(15, 93, 5)) # specify selection up to the number of features in the dataset (93)

# evaluate tuning w/ R^2
set.seed(main.seed)
tune.spls1.braincomp2.mbwin.R2 <- mixOmics::tune.spls(mb_data_brainint_win, brain.int.win.comp_scores$brain_int_win_comp2, ncomp= 2, 
                            test.keepX = list.keepX, 
                            validation = 'Mfold', 
                            folds = 10,
                            nrepeat = 100, 
                            progressBar = TRUE,
                            measure = 'R2') 

plot(tune.spls1.braincomp2.mbwin.R2) #highest R^2 (0.02) is with 2 components with 5 and 20 features

#how many components to keep w this?
choice.ncomp <- tune.spls1.braincomp2.mbwin.R2$choice.ncomp$ncomp #1
# how many x variables with 1 component? 5
choice.keepX <- tune.spls1.braincomp2.mbwin.R2$choice.keepX[1:2] #5 features

# specify final model with 2 components (best for model parsimony)
spls1.braincomp2.mb <- spls(X=mb_data_brainint, Y=brain.int.win.comp_scores$brain_int_win_comp2, ncomp = 2, keepX = choice.keepX, 
                    mode = "regression")

# extract list of features that were selected (each component is orthogonal)
selectVar(spls1.braincomp2.mb, comp = 1)$X$name #selected "coprostanoligenes" "Terrisporobacter"  "Enterococcus"      "Lactobacillus"     "Dialister"   
selectVar(spls1.braincomp2.mb, comp = 1)$X$value
plotLoadings(spls1.braincomp2.mb, 'X', comp=1) # positive is coprostanoligenes, Terrisporobacter, dialister; negative is enteroccocus, lactobacillus

# get the variable names and loading values
selectVar(spls1.braincomp2.mb, comp = 2)$X$name
mb_pattern3_loadings <- selectVar(spls1.braincomp2.mb, comp = 2)$X$value %>% rownames_to_column(var = "bacterium")
colnames(mb_pattern3_loadings) <- c("bacterium", "loading")
plotLoadings(spls1.braincomp2.mb, 'X', comp=2) 

# determine the variable importance (loading magnitude weighted by variance explained?)
vip.spls1.braincomp2.mb <- vip(spls1.braincomp2.mb)
vip.spls1.braincomp2.mb[selectVar(spls1.braincomp2.mb, comp=1)$X$name, 1] # 2 variables with VIP above 1 in comp 1
test <- as.data.frame(vip.spls1.braincomp2.mb[selectVar(spls1.braincomp2.mb, comp=2)$X$name, 2]) %>% rownames_to_column(var= "bacterium") # 10 variables with VIP above 1 in comp 1
colnames(test) <- c("bacterium", "VIP")

# merge loadings with vip
mb_pattern3_loadings <- mb_pattern3_loadings %>% left_join(test, by = "bacterium") %>% dplyr::mutate(loading = signif(loading, 2), VIP = signif(VIP, 2))

# write the output
write_csv(mb_pattern3_loadings, "tables/TableS3.csv")

plot(spls1.braincomp2.mb$variates$X, spls1.braincomp2.mb$variates$Y,
     xlab = 'X component', ylab = 'y component / scaled y') #looks like a nicer scatterplot than for comp1

cor(spls1.braincomp2.mb$variates$X, spls1.braincomp2.mb$variates$Y) # cor = .60

spls1.braincomp2.mb$prop_expl_var$X

# extract scores for mb component associated with brain comp2
braincomp2.mbwin <- as.data.frame(spls1.braincomp2.mb$variates$X) %>% 
  dplyr::rename(mb1_braincomp2 = comp1,
                mb2_braincomp2 = comp2) %>% 
  rownames_to_column(var="subID")
```

### Graph Microbe Loadings for SOFA Inter-Network Signature, and all Microbial Profiles Together
```{r}
# extract feature loadings
braincomp2_mb1 <- selectVar(spls1.braincomp2.mb, comp = 1)$X$value %>% rownames_to_column(var = "genus")
colnames(braincomp2_mb1)[2] <- "loading_C1"
braincomp2_mb2 <- selectVar(spls1.braincomp2.mb, comp = 2)$X$value %>% rownames_to_column(var = "genus")
colnames(braincomp2_mb2)[2] <- "loading_C2"

braincomp2_mb1$genus[braincomp2_mb1$genus == "coprostanoligenes"] <- "E. coprostanoligenes"

# create a factor that sorts the loadings by magnitude 
braincomp2_mb1$genus <- factor(braincomp2_mb1$genus, levels = braincomp2_mb1$genus[order(braincomp2_mb1$loading_C1, decreasing = TRUE)])

braincomp2_mb2$genus <- factor(braincomp2_mb2$genus, levels = braincomp2_mb2$genus[order(braincomp2_mb2$loading_C2, decreasing = TRUE)])

# graph loadings for second microbial profile by magnitude (2 loadings have VIP > 1)
mb_pattern2 <- ggplot(braincomp2_mb1 %>% slice_max(., n=2, order_by=abs(loading_C1)), aes(x = reorder(genus, loading_C1), y = loading_C1)) +
  geom_col(fill = '#fd988d') + 
  coord_flip() +
  ylab("Loading") + xlab("Genus") +
  labs(tag = "D") + 
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12), axis.title = element_text(size=16), plot.margin = margin(10, 10, 10, 10), plot.tag=element_text(size=14)) + 
  ggtitle("Microbial \n Profile 2") +
  theme(plot.title = element_text(hjust=0.5, size=14, face='bold'))

#ggsave('../../../../Users/Fran/conferences/bridget_australia/mbcomp1_brain2.jpg', width = 9, height=7, dpi=1000)

# graph 10 highest loadings by magnitude for third microbial profile (10 vars have VIP > 1)
mb_pattern3 <- 
  ggplot(braincomp2_mb2 %>% slice_max(., n=10, order_by=abs(loading_C2)), aes(x = reorder(genus, loading_C2), y = loading_C2)) +
  geom_col(fill = '#fd988d') + 
  coord_flip() +
  ylab("Loading") + xlab("Genus") +
  labs(tag = "E") + 
  theme_bw() +
  theme(axis.text.x = element_text(color="black", size=12), axis.text.y = element_text(color="black", size=12), axis.title = element_text(size=16), plot.margin = margin(10, 10, 10, 10), plot.tag=element_text(size=14)) +
  ggtitle("Microbial \n Profile 3") +
  theme(plot.title = element_text(hjust=0.5, size=14, face='bold'))

# save all 3 of the microbiome patterns together
library(patchwork)
mb_pattern1 + mb_pattern2 + mb_pattern3

#ggsave('figures/Figure2.png', mb_pattern1 + mb_pattern2 + mb_pattern3, width = 12, height=5, dpi=1000)
```

### Association of Microbial Patterns with Brain Signatures, controlling for covariates
Covariates are PUFA, dietary fat, fiber, GA, BW, and sex
```{r}
# add mb component scores to larger dataset with all covs and brain comp scores
all_data <- braincomp_alpha_covs %>% dplyr::left_join(braincomp1.mbwin, by = "subID") %>% left_join(braincomp2.mbwin, by = "subID") 

# comp1 mb, comp1 brain
braincomp1_mb <- lm(brain_int_noamyg_win_comp1 ~ mb_braincomp1 + sex_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(braincomp1_mb, robust=TRUE, show.std=TRUE, show.se=TRUE) #positive, B=.40, p = .038
braincomp1_mb_ncovs <- lm(brain_int_noamyg_win_comp1 ~ mb_braincomp1, data = all_data)
tab_model(braincomp1_mb_ncovs, robust=TRUE, show.std=TRUE, show.se=TRUE) #positive, B=0.51, p = .001

# comp1 mb, comp2 brain
braincomp2_mb1 <- lm(brain_int_noamyg_win_comp2 ~ mb1_braincomp2 + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(braincomp2_mb1, robust=TRUE, show.std=TRUE, show.se=TRUE) #positive, B=.55, p = .007
braincomp2_mb_ncovs <- lm(brain_int_noamyg_win_comp2 ~ mb1_braincomp2, data = all_data)
tab_model(braincomp2_mb_ncovs, robust=TRUE, show.std=TRUE) #positive, B=0.54, p < .001

# comp2 mb, comp2 brain
braincomp2_mb2 <- lm(brain_int_noamyg_win_comp2 ~ mb2_braincomp2 + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + GA_centered, data = all_data)
tab_model(braincomp2_mb2, robust=TRUE, show.std=TRUE, show.se=TRUE) #positive, B=.59, p = .012
braincomp2_mb_ncovs <- lm(brain_int_noamyg_win_comp2 ~ mb2_braincomp2, data = all_data)
tab_model(braincomp2_mb_ncovs, robust=TRUE, show.std=TRUE) #positive, B=0.55, p < .001
```

### Association of Microbial Patterns with Internalizing Symptoms, controlling for covariates (Total Effects)
```{r}
# N=43 for the below with covs, N=55 for without covs

# microbial profile 1
braincomp1_comp1mb_int <- lm(cbclintprobtot_y7_pos_boxcox ~ mb_braincomp1 + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(braincomp1_comp1mb_int, robust=TRUE, show.std=TRUE, show.se=TRUE) #ns, B = .28, p = .14

braincomp1_comp1mb_int_ncovs <- lm(cbclintprobtot_y7_pos_boxcox ~ mb_braincomp1, data = all_data)
tab_model(braincomp1_comp1mb_int_ncovs, robust=TRUE, show.std=TRUE, show.se=TRUE) # significant, B = .32, p = .01

# microbial profile 2
braincomp2_comp1mb_int <- lm(cbclintprobtot_y7_pos_boxcox ~ mb1_braincomp2 + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(braincomp2_comp1mb_int, robust=TRUE, show.std=TRUE, show.se=TRUE) # ns, B = .37, p = .06

braincomp2_comp1mb_int_ncovs <- lm(cbclintprobtot_y7_pos_boxcox ~ mb1_braincomp2, data = all_data)
tab_model(braincomp2_comp1mb_int_ncovs, robust=TRUE, show.std=TRUE) # ns, B = .22, p = .17

# microbial profile 3
braincomp2_comp2mb_int <- lm(cbclintprobtot_y7_pos_boxcox ~ mb2_braincomp2 + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(braincomp2_comp2mb_int, robust=TRUE, show.std=TRUE, show.se=TRUE) # ns, B = .07, p = .78

braincomp2_comp2mb_int_ncovs <- lm(cbclintprobtot_y7_pos_boxcox ~ mb2_braincomp2, data = all_data)
tab_model(braincomp2_comp2mb_int_ncovs, robust=TRUE, show.std=TRUE) # ns, B = .21, p = .14

# faith
faith_internalizing <- lm(cbclintprobtot_y7_pos_boxcox ~ faith_pd + sex_centered + GA_centered + BW_centered + deliv_mode + `Fat_%energy` + Fibre.per.1000kcal + `PUFA_%` + meanFD_centered, data = all_data)
tab_model(faith_internalizing, robust=TRUE, show.std=TRUE, show.se=TRUE)
```

## Mediation Models: Microbiome Features --> Brain Signatures --> Internalizing Symptoms 
Before using process run entire process.R script from Andrew Hayes to load the process() function
```{r}
# rename columns in all data with % in them to get function to work
all_data <- all_data %>% dplyr::mutate(
  Fat_perc_energy = `Fat_%energy`,
  PUFA_perc = `PUFA_%`
)

# **transformed var, process**
process(data=all_data, y="cbclintprobtot_y7_pos_boxcox", x="mb_braincomp1", m=c("brain_int_win_comp1"), cov=c("sex_centered", "GA_centered", "BW_centered", "deliv_mode", "Fat_%energy", "Fibre.per.1000kcal", "PUFA_%", "meanFD_centered"), model=4, seed=100770, stand=1) # significant a path, b path, n.s. positive direct and indirect effects (n.s. positive total effect given above)

process(data=all_data, y="cbclintprobtot_y7_pos_boxcox", x="mb1_braincomp2", m=c("brain_int_win_comp2"), cov=c("sex_centered", "GA_centered", "BW_centered", "deliv_mode", "Fat_%energy", "Fibre.per.1000kcal", "PUFA_%", "meanFD_centered"), model=4, seed=100770, stand=1) # significant a path, n.s. b path, n.s. positive direct and indirect effects (n.s. positive total effect)

process(data=all_data, y="cbclintprobtot_y7_pos_boxcox", x="mb2_braincomp2", m=c("brain_int_win_comp2"), cov=c("sex_centered", "GA_centered", "BW_centered", "deliv_mode", "Fat_%energy", "Fibre.per.1000kcal", "PUFA_%", "meanFD_centered"), model=4, seed=100770, stand=1) # significant a path and b path, significant positive indirect effect, n.s. negative direct effect (n.s. but positive total effect)


# faith as the x-variable iwth each brain component
process(data=all_data, y="cbclintprobtot_y7_pos_boxcox", x="faith_pd", m=c("brain_int_win_comp1"), cov=c("sex_centered", "GA_centered", "BW_centered", "deliv_mode", "Fat_%energy", "Fibre.per.1000kcal", "PUFA_%", "meanFD_centered"), model=4, seed=100770, stand=1) # n.s. direct and indirect, n.s. a path

process(data=all_data, y="cbclintprobtot_y7_pos_boxcox", x="faith_pd", m=c("brain_int_win_comp2"), cov=c("sex_centered", "GA_centered", "BW_centered", "deliv_mode", "Fat_%energy", "Fibre.per.1000kcal", "PUFA_%", "meanFD_centered"), model=4, seed=100770, stand=1) # positive a path, b path, n.s. indirect and direct effects
```

### Figure 1: Combined Brain Signatures and Microbial Patterns
```{r}
#combine the graphs generated in chunks above
library(patchwork)
plot_layout(brain_c1 + mb_pattern1 + plot_spacer() + brain_c2 + mb_pattern2 + mb_pattern3, nrow=2) 

ggsave('figures/Figure1_combined_aligned.jpg', width = 12, height=8, units="in", dpi=1000)
```

# Descriptives
## Table 1: Descriptive Statistics
```{r}
#analysis_sample <- all_data %>% dplyr::filter(!is.na(cbclintprobtot_y7))

# read in ethnicity
eth <- read_xlsx("../../data/pregnancy_mh/STAI_BDI_FormA515_20201218.xlsx") %>% dplyr::select(subID = SubjectID, ethnicity = mother_ethnicity)

# read in original sex, GA, meanFD, and BW measurements
birth <- read_xlsx("../../data/general_covariates/GA_weight_Ages_at_measures_FormA515_20210719.xlsx") %>% dplyr::select(subID = SubjectID, GA, BW = c_weight_birth)

sex <- read_xlsx("../../data/general_covariates/infant_sex.xlsx") %>% dplyr::select(subID = SubjectID, sex_binary) # 0 = female, 1 = male

meanFD <- read_csv("../../data/general_covariates/meanFD.csv") %>% dplyr::filter(age==6)

analysis_sample <- all_data %>% dplyr::filter(!is.na(cbclintprobtot_y7)) %>% dplyr::left_join(eth, by = "subID") %>% dplyr::left_join(birth, by = "subID") %>% dplyr::left_join(sex, by = "subID") %>% dplyr::left_join(meanFD, by = "subID")

# means and sds 
means_etc <- analysis_sample %>% dplyr::select(
  subID, # numbers in var names below are for ordering
  meanFD,
  cbclintprobtscore_y7,
  GA,
  BW,
  shannon_entropy,
  observed_features.x,
  pielou_evenness,
  faith_pd,
  `Protein_%energy`,
  `Fat_%energy`,
  `CHO_%energy`,
  Fibre.per.1000kcal,
  `SatFat_%`,
  `MUFA_%`,
  `PUFA_%`
) %>% pivot_longer(
  cols = !subID,
  names_to = "variable",
  values_to = "value"
) %>% group_by(variable) %>% 
  summarise(Mean = round(mean(value, na.rm=TRUE), digits=2), SD = round(sd(value, na.rm=TRUE), digits=2), Min = round(min(value, na.rm=TRUE), digits=2), Max = round(max(value, na.rm=TRUE),digits=2))

colnames(means_etc) <- c("Measure", "Mean", "SD", "Min", "Max")

write_csv(means_etc, "tables/Table1.csv")

# ns and percentages (ethnicity, sex, birth method, age stopped breastfeeding, )
analysis_sample %>% group_by(ethnicity) %>% summarise(n = n()) %>%
  mutate(freq = n / sum(n)) %>% arrange(desc(n))

analysis_sample %>% group_by(sex_binary) %>% summarise(n = n()) %>%
  mutate(freq = n / sum(n)) %>% arrange(desc(n)) # 0=female

analysis_sample %>% group_by(any_bf_months) %>% summarise(n = n()) %>%
  mutate(freq = n / sum(n)) %>% arrange(desc(n))

analysis_sample %>% group_by(deliv_mode_str) %>% summarise(n = n()) %>%
  mutate(freq = n / sum(n)) %>% arrange(desc(n))

# percent missing data check 
analysis_sample %>% dplyr::count(is.na(Fibre.per.1000kcal))
```

## Cronbach's Alpha for CBCL Internalizing Symptom Items in This Sample
```{r}
# merge cbcl raw items with anlytic sample ids
analysis_cbcl <- analysis_sample %>% dplyr::select(subID) %>% left_join(cbcl_raw_y7, by = "subID")

# recode the raw responses into numeric 
analysis_cbcl_clean <- analysis_cbcl %>% dplyr::mutate(
  across(contains("cbcl"), ~case_when(
    . == "Not True (as far as you know)" ~ 0,
    . == "Somewhat or Sometimes True" ~ 1,
    . == "Very True or Often True" ~ 2,
    . == "-9999" ~ NA
  ))
)

psych::alpha(analysis_cbcl_clean %>% dplyr::select(-subID)) # standardized alpha = 0.81
```

## Create Form B Dataset
Including child internalizing, symptoms, covariates, microbiome diversity, microbial patterns, brain signatures
```{r}
# select only the variables in all_data that you use
all_data_formb <- all_data %>% 
  dplyr::select(
    subID,
    sex_centered,
    GA_centered,
    BW_centered,
    meanFD_centered,
    any_bf_months,
    deliv_mode_str,
    `Fat_%energy`,
    Fibre.per.1000kcal,
    `PUFA_%`,
    `Protein_%energy`,
    `CHO_%energy`,
    `SatFat_%`,
    `MUFA_%`,
    cbclintprobtot_y7,
    faith_pd,
    observed_features = observed_features.x,
    shannon_entropy,
    pielou_evenness
  )

# add brain networks and microbes to the dataset
all_data_formb <- all_data_formb %>% left_join(mb_cbcl, by = "subID") %>% left_join(brain_cbcl_win, by = "subID") %>% dplyr::select(
  -contains("amyg")
)

write_csv(all_data_formb, "../../manuscripts/fran_jess_microbiome_brain/querdasi_mb-brain-int_formB (frq@ucla.edu)/querdasi_mb-brain-int_formB_data.csv")
```

